openstack/datadog_checks/openstack/openstack.py

```diff
@@ -876,39 +917,27 @@ def get_project_name_from_id(self, tenant_id):
             self.warning('Unable to get project name: {0}'.format(str(e)))
             raise e
 
-    def get_stats_for_single_server(self, server_id, tags=None):
+    def get_stats_for_single_server(self, server_details, tags=None):
         def _is_valid_metric(label):
             return label in NOVA_SERVER_METRICS or any(seg in label for seg in NOVA_SERVER_INTERFACE_SEGMENTS)
 
-        url = '{0}/servers/{1}'.format(self.get_nova_endpoint(), server_id)
-        headers = {'X-Auth-Token': self.get_auth_token()}
-        state = None
-        server_name = None
-        hypervisor_hostname = None
-        tenant_id = None
-        project_name = None
+        server_id = server.get('id')
+        state = server.get('status')
+        server_name = server.get('name')
+        hypervisor_hostname = server.get('OS-EXT-SRV-ATTR:hypervisor_hostname')
+        tenant_id = server.get('tenant_id')
+        project_name = self.get_project_name_from_id(tenant_id)
+
+        server_stats = {}
+        url = '{0}/servers/{1}/diagnostics'.format(self.get_nova_endpoint(), server_id)
```

Comment:
If you decide to keep servers in the cache in `SHUTOFF` state, remember that we should only be calling this if `state` is in a `DIAGNOSTICABLE` state (not much to worry about as the code is resilient to `InstancePowerOffFailure`).